### PR TITLE
Skip accountId and containerId check if blobId version is 1.

### PR DESCRIPTION
**Pull request by zzmao**
remove getAccountId() and getContainerId() in StoreKey.

Add isAccountContainerMatch() to StoreKey.

### CODE REVIEW

File: ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java [hunk 1]
```diff
@@ -272,11 +271,22 @@ public short getAccountId() {
...
   public short getContainerId() {
     return containerId;
   }
 
+  /**
+   * @return true if accountId and containerId in key match given accountId and containerId from store.
+   * Always return true if BlobId version is {@link #BLOB_ID_V1}.
+   */
+  @Override
+  public boolean isAccountContainerMatch(short accountId, short containerId) {
```
Comment (vgkholla) [inline ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java]: this function needs a test

File: ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobStoreRecoveryTest.java [hunk 2]
```diff
@@ -66,16 +66,19 @@ public String getLongForm() {
     return getID();
   }
 
-  @Override
   public short getAccountId() {
```
Comment (vgkholla) [inline ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobStoreRecoveryTest.java]: you can remove these two functions from here - they were added as part of the original change

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java [hunk 3]
```diff
@@ -578,70 +578,85 @@ public void deleteErrorCasesTest() {
   }
 
   /**
-   * Test DELETE authorization success for {@link BlobStore#delete(MessageWriteSet)}.
-   * If either accountId or containerId is -1, validation is skipped.
+   * Test DELETE with right accountId and containerId.
+   * If BlobId version is 1, accountId and containerId validation is skipped.
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: this piece of info is an implementation detail. This function should just be checking that auth succeeds when `isAccountContainerMatch()` returns `true` and fails when it returns `false`

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java [hunk 4]
```diff
@@ -578,70 +578,85 @@ public void deleteErrorCasesTest() {
...
    */
   @Test
   public void deleteAuthorizationSuccessTest() throws Exception {
     short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1};
     short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM)};
     for (int i = 0; i < accountIds.length; i++) {
-      MockId mockId = put(1, PUT_RECORD_SIZE, Utils.Infinite_Time, accountIds[i], containerIds[i]).get(0);
-      delete(
-          new MockId(mockId.getID(), Utils.getRandomShort(TestUtils.RANDOM), Utils.getRandomShort(TestUtils.RANDOM)));
+      // put and get version 1 blobId, acoountId and containerId check is skipped.
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: not required  - impl detail. Just check for success when `true` and failure when `false`

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java [hunk 5]
```diff
@@ -578,70 +578,85 @@ public void deleteErrorCasesTest() {
...
+      MockId mockId = put(1, PUT_RECORD_SIZE, Utils.Infinite_Time, (short) 1, accountIds[i], containerIds[i]).get(0);
+      delete(new MockId(mockId.getID(), (short) 1, Utils.getRandomShort(TestUtils.RANDOM),
+          Utils.getRandomShort(TestUtils.RANDOM)));
+      verifyDeleteFailure(mockId, StoreErrorCodes.ID_Deleted);
+    }
+
+    for (int i = 0; i < accountIds.length; i++) {
+      // put and get version 3 blobId, accountId and containerId check is skipped.
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: what does this mean? why is it skipped?

File: ambry-store/src/test/java/com.github.ambry.store/MockId.java [hunk 6]
```diff
@@ -23,16 +23,18 @@
 public class MockId extends StoreKey {
 
   private String id;
+  private short version = 3;
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/MockId.java]: no versions here - the impl need not be the same as `BlobId` - just needs to return `true`/`false` as required for `isAccountContainerMatch()`

File: ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java (hunk 1 above)
Comment (zzmao) [reply ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java]: fixed.

File: ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobStoreRecoveryTest.java (hunk 2 above)
Comment (zzmao) [reply ambry-messageformat/src/test/java/com.github.ambry.messageformat/BlobStoreRecoveryTest.java]: fixed.

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java (hunk 3 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: fixed.

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java (hunk 4 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: fixed.

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java (hunk 5 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: fixed.

File: ambry-store/src/test/java/com.github.ambry.store/MockId.java (hunk 6 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/MockId.java]: fixed.

File: ambry-tools/src/test/java/com.github.ambry/store/StoreCopierTest.java [hunk 7]
```diff
@@ -161,15 +161,16 @@ private void setupTestState() throws IOException, StoreException {
     src.start();
     try {
       deletedId = new MockId("deletedId");
-      addMessage(src, deletedId, Utils.Infinite_Time, false, deletedId.getAccountId(), deletedId.getContainerId(),
```
Comment (vgkholla) [inline ambry-tools/src/test/java/com.github.ambry/store/StoreCopierTest.java]: why was this changed?

**Comment by vgkholla** [comment]
LGTM - just one question on a test

File: ambry-tools/src/test/java/com.github.ambry/store/StoreCopierTest.java (hunk 7 above)
Comment (zzmao) [reply ambry-tools/src/test/java/com.github.ambry/store/StoreCopierTest.java]: Keep it same with before. https://github.com/linkedin/ambry/commit/dd1706531d573047ff99ce60132b0166809a2bcb#diff-744abfbfdb5da3ec75af6f62141777abL166
(Let me do one more change to make it exactly same with this.)

deleteId here is StoreKey, a cast need if we want to deleteId.getAccount().

Comment (zzmao) [reply ambry-tools/src/test/java/com.github.ambry/store/StoreCopierTest.java]: fixed.
